pkg/ingester/ingester_v2_test.go

```diff
@@ -1888,17 +1897,20 @@ func TestIngester_flushing(t *testing.T) {
 
 				i.FlushHandler(httptest.NewRecorder(), httptest.NewRequest("POST", "/flush", nil))
 
-				// Wait for compaction to finish.
-				test.Poll(t, 5*time.Second, true, func() interface{} {
+				// Flush handler only triggers compactions, but doesn't wait for them to finish. Let's wait for a moment, and then verify.
+				test.Poll(t, 1*time.Second, true, func() interface{} {
 					db := i.getTSDB(userID)
 					if db == nil {
 						return false
 					}
-
-					h := db.Head()
-					return h.NumSeries() == 0
+					return db.Head().NumSeries() == 0
 				})
 
+				// The above waiting only ensures compaction, waiting another second to register the Sync call.
+				time.Sleep(1 * time.Second)
```

Comment:
This sleep it similar to what we are doing above for other test cases. Additionally added polling for other cases.